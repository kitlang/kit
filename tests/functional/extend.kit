struct A {
    public var a: Int;
    public var b: Int;

    public static function f1() {
        puts("f1");
    }
}

extend A {
    public var c: Int;

    public static function f2() {
        puts("f2");
    }
}

struct StartsEmpty {}
extend StartsEmpty {
    public var nowItHasAField: Int;
}

function main() {
    var s = struct A {a: 1, b: 1, c: 1};
    printf("%i %i %i\n", s.a, s.b, s.c);
    A.f1();
    A.f2();
}
