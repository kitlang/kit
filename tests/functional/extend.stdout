1 1 1
f1
f2
